server/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -388,25 +429,32 @@ private boolean isOverTheLimit() {
     /**
      * Flush pending delete or index requests.
      */
-    public synchronized void flush() {
-        ensureOpen();
-        if (bulkRequest.numberOfActions() > 0) {
-            execute();
+    public void flush() {
+        try {
+            lock.lock();
+            ensureOpen();
+            if (bulkRequest.numberOfActions() > 0) {
+                execute();
+            }
+        }finally {
```

Comment:
```suggestion
        } finally {
```

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/41451/commits/a9e0f3bd8db3b7aa79ba2166895057c198475f56